index.js

```diff
@@ -81,7 +83,14 @@ module.exports = function (opts) {
 					deleteRequireCache(modId);
 				});
 			}
-			jasmine.addReporter(new SilentReporter(cb, errorOnFail));
+			jasmine.addReporter(new SilentReporter(function (error) {
+				if (error) {
+					cb(error);
+				} else {
+					self.emit('end');
+					cb();
```

Comment:
I am still pretty bad with streams but getting better, I hope.  A Writable stream is completely different than a Transform stream.  There are 4 types of streams: Readable (must start a pipeline), Writable (must end a pipeline), Duplex (Readable + Writable but necessarily coupled), and Transform (Readable & Writable stream that are coupled).  I think the events (for a Transform stream) are best summed up by the Transform stream docs:
> Events: 'finish' and 'end'#
> The 'finish' and 'end' events are from the parent Writable and Readable classes respectively. The 'finish' event is fired after stream.end() is called and all chunks have been processed by stream._transform(), 'end' is fired after all data has been output which is after the callback in stream._flush() has been called.

A transform stream can't end the pipeline but since it's inception, gulp has recommended this pattern to users (either directly or indirectly through plugins due to miscommunication or incorrect understanding of streams).  This was handled by adding a "drain" (or "resume") step to the task execution, which happens if you return the stream in both gulp 3.x and 4.0.  On top of that, vinyl-fs changed at some point to use a Transform stream for it's `.dest()` method and streams would stop processing at 16 files, unless returned and drained; however, we've recently gone through the paces of making a Transform stream act sometimes as a Writable and sometimes as a Transform stream (depending on whether it was piped).  I do **not** recommend doing this because it was an absolute hassle and I'm not completely sure if it works in every case.